azurerm/resource_arm_cosmosdb_mongo_collection.go

```diff
@@ -118,20 +120,27 @@ func resourceArmCosmosDbMongoCollectionCreateUpdate(d *schema.ResourceData, meta
 	account := d.Get("account_name").(string)
 	database := d.Get("database_name").(string)
 	throughput := d.Get("throughput").(int)
+	dbHasThroughputConfigured := throughput > 0
 
-	if features.ShouldResourcesBeImported() && d.IsNewResource() {
-		existing, err := client.GetMongoDBCollection(ctx, resourceGroup, account, database, name)
-		if err != nil {
-			if !utils.ResponseWasNotFound(existing.Response) {
-				return fmt.Errorf("Error checking for presence of creating Cosmos Mongo Collection %s (Account %s, Database %s): %+v", name, account, database, err)
-			}
-		} else {
-			id, err := azure.CosmosGetIDFromResponse(existing.Response)
+	createUpdateOptions := map[string]*string{}
+
+	if d.IsNewResource() {
+		if features.ShouldResourcesBeImported() {
+			existing, err := client.GetMongoDBCollection(ctx, resourceGroup, account, database, name)
 			if err != nil {
-				return fmt.Errorf("Error generating import ID for Cosmos Mongo Collection %s (Account %s, Database %s)", name, account, database)
-			}
+				if !utils.ResponseWasNotFound(existing.Response) {
+					return fmt.Errorf("Error checking for presence of creating Cosmos Mongo Collection %s (Account %s, Database %s): %+v", name, account, database, err)
+				}
+			} else {
+				id, err := azure.CosmosGetIDFromResponse(existing.Response)
+				if err != nil {
+					return fmt.Errorf("Error generating import ID for Cosmos Mongo Collection %s (Account %s, Database %s)", name, account, database)
+				}
 
-			return tf.ImportAsExistsError("azurerm_cosmosdb_mongo_collection", id)
+				return tf.ImportAsExistsError("azurerm_cosmosdb_mongo_collection", id)
+			}
+		} else if dbHasThroughputConfigured {
+			createUpdateOptions["throughput"] = utils.String(strconv.Itoa(throughput))
 		}
```

Comment:
I think we'll need to change this to

```suggestion
		if hasThroughput {
			createUpdateOptions["throughput"] = utils.String(strconv.Itoa(throughput))
		}
```

Reply:
looks correct, but don't we update the resource when executing "terraform import" with the current logic?

Reply:
manual tests did not show what I was assuming so I guess it's fine